src/box/tuple_format.c

```diff
@@ -111,12 +111,33 @@ tuple_format_cmp(const struct tuple_format *format1,
 		if (field_a->is_key_part != field_b->is_key_part)
 			return (int)field_a->is_key_part -
 				(int)field_b->is_key_part;
-		/*
-		 * Format comparison is used for non-space formats only,
-		 * so there must be no constraints in both formats.
-		 */
-		assert(field_a->constraint_count == 0);
-		assert(field_b->constraint_count == 0);
+		if (field_a->sql_default_value_expr !=
+		    field_b->sql_default_value_expr)
+			return (int)(intptr_t)field_a->sql_default_value_expr -
+			       (int)(intptr_t)field_b->sql_default_value_expr;
```

Comment:
1. Please add a comment explaining why comparing pointers is OK.
2. The cast is incorrect. It can result in equality (0) even for different pointers.
3. Please add a comment explaining why we can't compare the format MsgPack data.

Reply:
Fixed.